cpp/test/cpu_operations_test/transpose_test.cc

```diff
@@ -1,66 +1,84 @@
-// This file tests the cpu_operations.cc Transpose() function by comparing a
-// Transposed Eigen::Matrix and a Nice::Matrix
-// Tests are made for integers and floats by using a template type test
-// Behavior with oddly shaped matrices (oddshape tests) is also tested with
-// Matrices of size (1,4) and (0,2)
+// The MIT License (MIT)
+//
+// Copyright (c) 2016 Northeastern University
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// This file tests the CpuOperations::Transpose() function by checking to
+// See if a matrix passed is transposed in the test IsTransposed
+// A transposed Nice matrix is compared to a transposed Eigen Matrix in
+// Transpose Eigen
+// Behavior with oddly shaped matrices is also tested with test DifferentShapes
+// And TransposeZeroRows
+// All tests are made using a templated test fixture which attempts
+// Integer, float, and double data types
 
-#include <iostream>
 #include <stdio.h>
+#include <iostream>
 #include "Eigen/Dense"
-#include <gtest/gtest.h>
-#include "cpu_operations.h"
-#include "cpu_operations.cc"  // Included to reference the templated class
-#include "matrix.h"  // This is included for testing purposes
+#include "gtest/gtest.h"
+#include "include/cpu_operations.h"
+#include "include/matrix.h"
 
 // This is a template test fixture class containing Eigen and NICE matrices
 template<class T>  // Template
 class MyTest : public ::testing::Test {  // Inherits from testing::Test
- public:
-
-  Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> _matrix_eigen;  // Public members
+ public:  // Members must be public to be accessed by tests
+  Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> _matrix_eigen;
```

Comment:
Does having a leading underscore in the variable name follow the naming convention?